src/terminal.c

```diff
@@ -4292,6 +4292,67 @@ handle_call_command(term_T *term, channel_T *channel, listitem_T *item)
 	ch_log(channel, "Calling function %s failed", func);
 }
 
+/*
+ * URL decoding (also know as Percent-encoding).
+ *
+ * Note this function currently is only used for decoding shell's 
+ * OSC 7 escape sequence which we can assume all bytes are valid 
+ * UTF-8 bytes. Thus we don't need to deal with invalid UTF-8 
+ * encoding bytes like 0xfe, 0xff. 
+ */
+static size_t url_decode(const char *src, const size_t len, char_u *dst)
+{
+  size_t i = 0, j = 0;
+
+  while (i < len)
+  {
+    if(src[i] == '%' && i + 2 < len)
+    {
+      char buffer[3] = {src[i+1], src[i+2], 0};
+      dst[j] = strtol(buffer, NULL, 16);
+      j++;
+      i += 3;
+    } else {
+      dst[j] = src[i];
+      i++;
+      j++;
+    }
+  }
+  dst[j] = '\0';
+  return j;
+}
+
+/*
+ * Sync terminal buffer's cwd with shell's pwd with the help
+ * of OSC 7.
+ *
+ * The OSC 7 sequence has the format of 
+ * "\033]7;file://HOSTNAME/CURRENT/DIR\033\\" 
+ * and what VTerm provides via VTermStringFragment is
+ * "file://HOSTNAME/CURRENT/DIR"
+ */
+static void syn_cwd(VTermStringFragment *frag)
+{
+    int       offset = 7; // len of "file://" is 7
+    char      *pos = (char *)frag->str + offset;
+
+    // remove HOSTNAME to get PWD
+    while (*pos != '/' && offset < frag->len) {
+        offset += 1;
+        pos += 1;
+    }
+
+    if (offset >= frag->len) {
+        semsg("Failed to extract PWD from %s, please check your shell's config related OSC 7", frag->str);
+        return;
+    }
+
+    char_u *new_dir = malloc(frag->len - offset + 1);
```

Comment:
This variable should be declared at the top of the file, that'd fix the `C2275` error on Appveyor.

Reply:
Thanks for providing the fix!